homeassistant/components/media_player/kodi.py

```diff
@@ -130,6 +130,8 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
         password=config.get(CONF_PASSWORD),
         turn_off_action=config.get(CONF_TURN_OFF_ACTION), websocket=websocket)
 
+    key = 'media_player.' + slugify(config.get(CONF_NAME))
+    hass.data[DATA_KODI][key] = entity
     async_add_devices([entity], update_before_add=True)
```

Comment:
Ah, I missed that. In that case then yeah, going back to the list seems to make the most sense.